core/file_walker.go

```diff
@@ -1,50 +1,70 @@
 package core
 
 import (
-	"fmt"
+	"log"
 	"os"
 	"path/filepath"
 )
 
 type File struct {
 	Name  string
+	Dir   *File
 	Size  int64
-	IsDir bool
+	IsDir bool // TODO(leon): IsDir() => len(f.Files) > 0
```

Comment:
Because we are calling `core.PruneTree` and removing files/folders that are smaller then certain limit, folder without any files in `Files` property can exist (https://github.com/viktomas/godu/blob/master/core/processor.go#L9)